lib/parser.js

```diff
@@ -253,4 +254,28 @@ async function customChannelsOperations(parsedJSON, asyncapiYAMLorJSON, initialF
     }));
   });
   await Promise.all(promisesArray);
-}
\ No newline at end of file
+}
+
+/**
+ * Triggers additional operations on the AsyncAPI components/messages like traits application, validation and conversion
+ *
+ * @private
+ * 
+ * @param  {Object} parsedJSON parsed AsyncAPI document
+ * @param  {String} asyncapiYAMLorJSON AsyncAPI document in string
+ * @param  {String} initialFormat information of the document was originally JSON or YAML
+ * @param  {Object} options Configuration options.
+ */
+async function customMessagesOperations(parsedJSON, asyncapiYAMLorJSON, initialFormat, options) {
+  const promisesArray = [];
+
+  Object.entries(parsedJSON.components.messages).forEach(([messageName, message]) => {
+    if (options.applyTraits) {
+      applyTraits(message);
+    }
+    const pathToPayload = `/components/messages/${messageName}/payload`;
+    promisesArray.push(validateAndConvertMessage(message, asyncapiYAMLorJSON, initialFormat, parsedJSON, pathToPayload));
```

Comment:
To your question about `validateAndConvertMessage` function. In AscyncAPI spec you can define in which format you want describe messages, in jsonSchema, avro, raml (at the moment officially we only support them) and we validate and parse message in this function based on format, so your usage is good :) 
Ref in doc: https://github.com/asyncapi/asyncapi/blob/master/versions/2.0.0/asyncapi.md#messageObject